studiopress/pattern-manager
# Rough draft of patternception POC

This branch is a POC of how patterns-inside-patterns could work:

### How to test
1. Whenever you want to use a pattern inside another pattern, use the PM:Pattern block:
<img width="425" alt="Screen Shot 2023-03-27 at 4 33 48 PM" src="https://user-images.githubusercontent.com/7538525/228060072-3f73ec8e-6337-4051-8d17-69e844280676.png">
2. Type the name of the pattern you wish to use (this should instead be a list of patterns in the theme to choose from)
<img width="1393" alt="Screen Shot 2023-03-27 at 4 35 04 PM" src="https://user-images.githubusercontent.com/7538525/228060415-fb538598-2969-4f6e-89b8-9d340dbb9330.png">
3. When you save the pattern, the block code saved to the file will use the 
<img width="325" alt="Screen Shot 2023-03-27 at 4 36 15 PM" src="https://user-images.githubusercontent.com/7538525/228060649-280e04ca-7a62-42fe-bb68-b0226872c707.png">
4. When you load up that pattern again, it will be using the PM block.


### To do:
- Make the "Edit This Pattern" button save the current pattern, then open the pattern in question.
<img width="1173" alt="Screen Shot 2023-03-27 at 4 38 20 PM" src="https://user-images.githubusercontent.com/7538525/228061058-dcd834d1-a088-4998-8c24-8edd61281993.png">
- Possibly, make our pattern block show a preview of the blocks inside that pattern (but they should **NOT be editable**)
